File: 05-multiplication-table/mult_table.f90
! Fortran program to print Multiplication tables
! Written by M.V.Harish Kumar on 09-05-2024
program mult_table
    implicit none
    integer :: n, t
    
    print '(a$)', "Enter the no. to generate tables: "
    read(*,*) n
    print '(a$)', "Enter the no. till which to generate: "
    read(*,*) t
    call gen_tables(n, t)

contains
    subroutine gen_tables(number, times)
        implicit none
        integer, intent(in) :: number, times
        integer :: i

        print '(i0,a)', number, ' Tables'
        do i = 1, times
            print '(i0,a,i0,a,i0)', number, ' x ', i, ' = ', number*i
        end do
    end subroutine gen_tables
end program mult_table

File: 05-multiplication-table/mult_table.ml
(* Ocaml program to print multiplication tables *)
(* Written by M.V.Harish Kumar on 28-05-2024 *)
open Printf
let gen_tables n t =
  printf "%d Tables\n\n" n;
  for i = 1 to t do
    printf "%d x %d = %d\n" n i (n*i)
  done

let () =
  print_string "Enter the no. to generate tables: ";
  let num = read_int () in
    print_string "Enter the no. till which to generate: ";
    let times = read_int () in
      gen_tables num times